Adapt README to the new extension settings (#227)
alanz/vscode-hie-server: README.md

Client interface to the Language Server Protocol server for Haskell, as provided by the [Haskell IDE Engine](https://github.com/haskell/haskell-ide-engine), [ghcide](https://github.com/digital-asset/ghcide) or the new [Haskell Language Server](https://github.com/haskell/haskell-language-server).
Check the [requirements](#user-content-requirements) for dependencies.

**It is still under development!** If you want to help, get started by reading [Contributing](https://github.com/alanz/vscode-hie-server/blob/master/Contributing.md) for more details.

## Requirements

The language client requires you to manually install at least one of:

* [Haskell IDE Engine](https://github.com/haskell/haskell-ide-engine#installation): It was the unique haskell LSP server supported by this extension until version `0.40.0`. It is stable and functional but it will be replaced sooner or later by the new Haskell Language Server (see below).
* [ghcide](https://github.com/digital-asset/ghcide#install-ghcide): A fast and reliable LSP server with the [main basic features](https://github.com/digital-asset/ghcide#features). Supported since the `0.40.0` version of the extension.

> * [Haskell language server](https://github.com/haskell/haskell-language-server#installation): The future succesor of haskell-ide-engine. Its goal is to add features like formatting and refactor on top of ghcide, via custom plugins. It is still under heavy development and it does not have all the features of haskell-ide-engine, so use at your own risk! It is supported since the `0.40.0` version of the extension.

Reviewer: ```suggestion
* [Haskell language server](https://github.com/haskell/haskell-language-server#installation): The future successor of haskell-ide-engine. It is still under heavy development and it does not have all the features of haskell-ide-engine, yet, so use at your own risk! It is supported since the `0.40.0` version of the extension.
```

I dont think an introdcution to HLS's concept is important for the vscode-plugin.

If the liquid haskell executable is installed, enable using it to
process haskell files on save.

```json
"languageServerHaskell.liquidOn": true,
```

### HIE Variant

> Since `0.40` the extension no longer uses the wrapper. It has been replaced by a selection over

Reviewer: Wait what? It doesnt? Is always `hie`/`ghcide`/`hls` selected now or the respective wrapper?

Author: Mmm, the wrapper was removed, the selector is used to know what executable should be called
In fact the wrapper could be replaced with the hie executable field, no?

Reviewer: Ok, just wanted to make sure. I probably dont know what the purpose of the wrapper was then? 
To me it sounds like `hie-wrapper` and `haskell-language-server-wrapper` are obsolete now.

Author: Mmm but we need to switch between hie and hls versions to get the appropiate one for the project ghc, no?

Reviewer: I think so, too. We need the `-wrapper` executables, right?  (I dont use them at the moment)
This documentation sentence is a bit confusing to me.

Author: Oh, i see, the wrappers executables are still needed although the fields to set the custom wrapper are not, i will rephrase

Author: @fendor i've removed the reference to wrapper entirely cause, to be precise, the new field does not replace the custom wrapper

Reviewer: Nice! Thanks!

the three supported language servers: `haskell-ide-engine`, `ghcide` and `haskell-language-server`.

> The default one is `haskell-ide-engine`, although it will be changed by `haskell-language-server`
> when it would be stable enough.

Reviewer: ```suggestion
The default one is `haskell-ide-engine`, although it will be changed to `haskell-language-server`
when it will be stable enough.
```

The extension will look for the executable in `$PATH` and it will call it with the appropiate params

> depending on the extension settings. However, not all extension settings could be applied to all the language

Reviewer: ```suggestion
depending on the extension settings. However, not all extension settings can be applied to all the language
```

servers:

* `haskell-ide-engine`: It supports all of them.
* `ghcide`: It does not support any of them.

> * `haskell-language-server`: It only supports for now the log related settings: `Log File` and `Trace:server`. The goal is it will support all the same settings as `haskell-ide-engine`.

Reviewer: ```suggestion
* `haskell-language-server`: For now it only supports the log related settings: `Log File` and `Trace:server`. The goal is to support the same settings as `haskell-ide-engine`.
```

> $ stack --stack-yaml=<stack.yaml you used to build hie> exec hoogle generate

Reviewer: Make clear that this can only be done in Haskell IDE Engine and to use the appropriate target in the installation script, maybe?

Author: Well, in fact that command does not fail for ghcide and hls. The hls script failed cause it does a `stack build hoogle && stack run hoogle generate` and `stack build` requires the package as dependency or target.
It is noted that both will not use the hoogle database so maybe it would be enough as is?
